test/lightning/accounts/user_notifier_test.exs

```diff
@@ -76,53 +77,249 @@ defmodule Lightning.Accounts.UserNotifierTest do
       )
     end
 
-    test "deliver_project_digest/4" do
-      user = %User{email: "real@email.com"}
+    test "build_digest_url/3" do
+      workflow = Lightning.WorkflowsFixtures.workflow_fixture(name: "Workflow A")
+      start_date = Timex.now()
+      end_date = Timex.now() |> Timex.shift(days: 2)
+
+      query =
+        UserNotifier.build_digest_url(workflow, start_date, end_date)
+        |> URI.decode_query()
```

Comment:
Please add a test to verify the path that is returned here is correct.

Use `URI.parse/1` and match on the `:path` key to check that the url has the correct path.

Oh and just thinking out loud here, you might want to use the `:rfc3986` option when decoding the query. Just try it. The default decoding pattern for decoding queries is `:www_form` which is slightly different to URI query params.

See [`decode_query/3`](https://hexdocs.pm/elixir/1.14.4/URI.html#decode_query/3)

Reply:
Done. Thanks for the recommendation